File: PMSUI/src/models/product.ts
import { Brand } from "./brand";
import { Variant } from "./variant";

export class Product {
  id!: number;
  name!: string;
  type!: ProductType;
  brandId!: number;
  brand!: Brand;
  variants!: Variant[];
}
export enum ProductType {
  Mug = 'Mug',
  Jug = 'Jug',
  Cup = 'Cup'
}
export enum Size {
  Small = 'Small',
  Medium = 'Medium',
  Large = 'Large'
}

File: Core/Models/Product.cs
using Core.Enums;
using System.ComponentModel.DataAnnotations;

namespace Core.Models
{
    public class Product
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public ProductType Type { get; set; }
        public int BrandId { get; set; }

        public virtual Brand Brand { get; set; }
        public virtual ICollection<Variant> Variants { get; set; }
    }
}
